### DISCUSSION TITLE
Constraint violation

### DISCUSSION THREAD

**Discussion by HippoBloke** [jonaswinkler/paperless-ng#863] [Support]
Hi all

I've noticed a couple of constraint violations during the consume process recently. Is there anyway I can perform a database sanity check ?

Could there be two consumption processes running so the checksum gets duplicated ?

e.g. 

Kia service plan.pdf: The following error occured while consuming Kia service plan.pdf: UNIQUE constraint failed: documents_document.checksum : Traceback (most recent call last):
File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
return self.cursor.execute(sql, params)
File "/usr/local/lib/python3.7/site-packages/django/db/backends/sqlite3/base.py", line 413, in execute
return Database.Cursor.execute(self, query, params)
sqlite3.IntegrityError: UNIQUE constraint failed: documents_document.checksum

Thanks
Hippo

**Comment by amenk** [comment]
Is this during document consumption? I believe it means that paperless-ng tried to consume the exact same document twice.

**Reply by HippoBloke** [reply]
It is. Seems the delete process is not fast enough for the consume process.
